app/battle/fight.py

```diff
@@ -0,0 +1,13 @@
+def battle(battlers: dict) -> None:
+    # 1 Lancelot vs Mordred:
+    battlers["lancelot"]["hp"] -= battlers["mordred"]["power"] - battlers["lancelot"]["protection"]
+    battlers["mordred"]["hp"] -= battlers["lancelot"]["power"] - battlers["mordred"]["protection"]

```

Comment:
Create additional function for this because code is duplicated twice you can just pass dict and two names